Running `electron-builder --appimage` with toolset 1.0.2 on Ubuntu 25.10 fails

Recently trying to add AppImage support to Lemonade (https://github.com/lemonade-sdk/lemonade/pull/1079) and notice that if the AppImage is built on an Ubuntu 25.10 userspace then we get a failure during build.

Error:
```
Exit code: 1. Command failed: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs /home/supermario/src/lemonade/build/app-appimage/__appImage-x64 /home/supermario/src/lemonade/build/app-appimage/Lemonade-9.3.4-x86_64.AppImage -offset 944632 -all-root -noappend -no-p 1e39f55a1482e41611a25825129d533f2d3b23117f519db70092a7270369f1c0 (sha256 hash) -quiet -no-xattrs -no-fragments
basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
                                basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
```

Full log:
```
FAILED: CMakeFiles/appimage /home/supermario/src/lemonade/build/CMakeFiles/appimage 
cd /home/supermario/src/lemonade/build && /usr/bin/cmake -E echo Copying\ Electron\ app\ source\ to\ AppImage\ build\ directory... && /usr/bin/cmake -E remove_directory /home/supermario/src/lemonade/build/app-appimage-src && /usr/bin/cmake -E make_directory /home/supermario/src/lemonade/build/app-appimage-src && /usr/bin/cmake -E copy_directory /home/supermario/src/lemonade/src/app /home/supermario/src/lemonade/build/app-appimage-src && /usr/bin/cmake -E echo Setting\ version\ to\ 9.3.4... && /usr/bin/cmake -E chdir /home/supermario/src/lemonade/build/app-appimage-src /usr/bin/npm version 9.3.4 --no-git-tag-version --allow-same-version && /usr/bin/cmake -E echo Installing\ npm\ dependencies\ for\ AppImage\ build... && /usr/bin/cmake -E chdir /home/supermario/src/lemonade/build/app-appimage-src /usr/bin/npm ci && /usr/bin/cmake -E echo Building\ AppImage\ with\ electron-builder... && /usr/bin/cmake -E chdir /home/supermario/src/lemonade/build/app-appimage-src /usr/bin/npm run build:renderer:prod && /usr/bin/cmake -E chdir /home/supermario/src/lemonade/build/app-appimage-src /usr/bin/npm exec -- electron-builder --linux AppImage --config.directories.output=/home/supermario/src/lemonade/build/app-appimage && /usr/bin/cmake -E echo AppImage\ build\ complete.\ Output\ in\ /home/supermario/src/lemonade/build/app-appimage
Copying Electron app source to AppImage build directory...
Setting version to 9.3.4...
v9.3.4
Installing npm dependencies for AppImage build...
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@electron/rebuild@4.0.3',
npm WARN EBADENGINE   required: { node: '>=22.12.0' },
npm WARN EBADENGINE   current: { node: 'v20.19.4', npm: '9.2.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'node-abi@4.26.0',
npm WARN EBADENGINE   required: { node: '>=22.12.0' },
npm WARN EBADENGINE   current: { node: 'v20.19.4', npm: '9.2.0' }
npm WARN EBADENGINE }
npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm WARN deprecated rimraf@2.6.3: Rimraf versions prior to v4 are no longer supported
npm WARN deprecated glob@7.2.3: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me
npm WARN deprecated glob@7.2.3: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me
npm WARN deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm WARN deprecated node-domexception@1.0.0: Use your platform's native DOMException instead
npm WARN deprecated glob@10.5.0: Old versions of glob are not supported, and contain widely publicized security vulnerabilities, which have been fixed in the current version. Please update. Support for old versions may be purchased (at exorbitant rates) by contacting i@izs.me

added 740 packages, and audited 741 packages in 6s

141 packages are looking for funding
  run `npm fund` for details

13 vulnerabilities (2 low, 10 moderate, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
Building AppImage with electron-builder...

> lemonade@9.3.4 build:renderer:prod
> webpack --mode production

assets by chunk 1.15 MiB (auxiliary name: main) 65 assets
assets by path *.ts 6.58 KiB 18 assets
assets by path *.map 7.72 KiB 18 assets
assets by path utils/ 17.2 KiB
  assets by path utils/*.ts 9.82 KiB 6 assets
  assets by path utils/*.map 7.39 KiB 6 assets
assets by path hooks/ 3.57 KiB
  assets by path hooks/*.map 1.75 KiB 3 assets
  assets by path hooks/*.ts 1.82 KiB 3 assets
assets by path recipes/ 7.46 KiB
  assets by path recipes/*.ts 4.3 KiB 2 assets
  assets by path recipes/*.map 3.16 KiB 2 assets
asset renderer.bundle.js 1.78 MiB [emitted] [minimized] (name: main) 1 related asset
asset index.html 583 bytes [emitted]
runtime modules 2.63 KiB 8 modules
javascript modules 3.48 MiB 260 modules
asset modules 1.15 MiB (asset) 2.93 KiB (javascript)
  modules by path ./node_modules/ 1.03 MiB (asset) 2.46 KiB (javascript)
    modules by path ./node_modules/katex/dist/fonts/*.woff2 254 KiB (asset) 840 bytes (javascript) 20 modules
    modules by path ./node_modules/katex/dist/fonts/*.woff 296 KiB (asset) 840 bytes (javascript) 20 modules
    modules by path ./node_modules/katex/dist/fonts/*.ttf 502 KiB (asset) 840 bytes (javascript) 20 modules
  modules by path ./assets/ 129 KiB (asset) 210 bytes (javascript)
    ./assets/favicon.ico 124 KiB (asset) 42 bytes (javascript) [built] [code generated]
    ./assets/logo.svg 4.78 KiB (asset) 42 bytes (javascript) [built] [code generated]
    ./assets/chevron-down-gray.svg 235 bytes (asset) 42 bytes (javascript) [built] [code generated]
    ./assets/chevron-down-light.svg 235 bytes (asset) 42 bytes (javascript) [built] [code generated]
    ./assets/chevron-down-white.svg 235 bytes (asset) 42 bytes (javascript) [built] [code generated]
  data:image/svg+xml,%3Csvg xmlns=%27.. 273 bytes [built] [code generated]
webpack 5.102.1 compiled successfully in 7141 ms
  • electron-builder  version=26.8.1 os=6.19.0-rc4-00064-gc9adc6784e1b
  • loaded configuration  file=package.json ("build" field)
  • executing @electron/rebuild  electronVersion=39.2.4 arch=x64 buildFromSource=false workspaceRoot=/home/supermario/src/lemonade/build/app-appimage-src projectDir=./ appDir=./
  • installing native dependencies  arch=x64

  • completed installing native dependencies
  • packaging       platform=linux arch=x64 electron=39.2.4 appOutDir=/home/supermario/src/lemonade/build/app-appimage/linux-unpacked
  • searching for node modules  pm=npm searchDir=/home/supermario/src/lemonade/build/app-appimage-src
  • duplicate dependency references  dependencies=["call-bind-apply-helpers@1.0.2","es-object-atoms@1.1.1","hasown@2.0.2","hasown@2.0.2","form-data@4.0.5"]
  • building        target=AppImage arch=x64 file=/home/supermario/src/lemonade/build/app-appimage/Lemonade-9.3.4-x86_64.AppImage
  ⨯ failed to build AppImage  error=/home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs process failed 1
Exit code:
1
Output:
Exit code: 1. Command failed: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs /home/supermario/src/lemonade/build/app-appimage/__appImage-x64 /home/supermario/src/lemonade/build/app-appimage/Lemonade-9.3.4-x86_64.AppImage -offset 944632 -all-root -noappend -no-p 1e39f55a1482e41611a25825129d533f2d3b23117f519db70092a7270369f1c0 (sha256 hash) -quiet -no-xattrs -no-fragments
basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
                                basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)

  ⨯ /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs process failed 1
Exit code:
1
Output:
Exit code: 1. Command failed: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs /home/supermario/src/lemonade/build/app-appimage/__appImage-x64 /home/supermario/src/lemonade/build/app-appimage/Lemonade-9.3.4-x86_64.AppImage -offset 944632 -all-root -noappend -no-p 1e39f55a1482e41611a25825129d533f2d3b23117f519db70092a7270369f1c0 (sha256 hash) -quiet -no-xattrs -no-fragments
basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)

basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
  failedTask=build stackTrace=Error: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs process failed 1
Exit code:
1
Output:
Exit code: 1. Command failed: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/mksquashfs /home/supermario/src/lemonade/build/app-appimage/__appImage-x64 /home/supermario/src/lemonade/build/app-appimage/Lemonade-9.3.4-x86_64.AppImage -offset 944632 -all-root -noappend -no-p 1e39f55a1482e41611a25825129d533f2d3b23117f519db70092a7270369f1c0 (sha256 hash) -quiet -no-xattrs -no-fragments
basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     basename: /home/supermario/.cache/electron-builder/appimage@1.0.2/appimage-tools-runtime-20251108-1fgqg/linux/x64/lib/libm.so.6: version `GLIBC_2.29' not found (required by basename)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         at /home/supermario/src/lemonade/build/app-appimage-src/node_modules/builder-util/src/util.ts:155:18
    at ChildProcess.exithandler (node:child_process:430:5)
    at ChildProcess.emit (node:events:524:28)
    at maybeClose (node:internal/child_process:1104:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
ninja: build stopped: subcommand failed.
```

--- a/.changeset/shiny-rivers-turn.md
+++ b/.changeset/shiny-rivers-turn.md
@@ -1,0 +1,5 @@
+---
+"appimage": patch
+---
+
+fix(appimage): exclude libm from AppImage build process

--- a/packages/appimage/assets/build-appimage.sh
+++ b/packages/appimage/assets/build-appimage.sh
@@ -310,7 +310,7 @@ find "$ARCH_OUTPUT_DIR" -type f -perm -111 | while read -r binary; do
             
             # Skip system libraries
             case "$libname" in
-                libc.so.*|ld-linux*.so.*|libpthread.so.*|librt.so.*|libdl.so.*)
+                libc.so.*|libm.so.*|libm-*.so|ld-linux*.so.*|libpthread.so.*|librt.so.*|libdl.so.*)
                     continue
                 ;;
             esac

